### PR TITLE
allow argv2 for deeplinks

### PR DESCRIPTION
**Pull request by heronhaye**
Addresses https://github.com/keybase/client/issues/18925.

@eli-schwartz, @taigrr, please check out this PR.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
I rebuilt my distro package for keybase/kbfs/keybase-gui 4.3.1 with commit c8f65d0e2fd868327c8f275fc32177e6d1cb089f applied. Then I tried to start my keybase-gui wrapper script with a keybase:// link, and it worked as expected. So this PR looks good to me. Thanks!

**Comment by heronhaye** [comment]
@eli-schwartz I noticed there's another codepath here. Technically, this should work even if Keybase isn't running already. But I was unable to figure out how to pass in the argument to the electron binary from run_keybase (or in your case keybase-gui) when starting the GUI via systemd. So this flow doesn't work, but it will if you manually run `electron path-to-app keybase://xxx`. 

In any case, I added the keybase:// handler and argv2 handling to that flow as well. If you manage to figure out how to implement it, let us know!

File: shared/desktop/app/node.desktop.tsx [hunk 1]
```diff
@@ -161,9 +169,17 @@ const createMainWindow = () => {
       // stash a startupURL to be dispatched when we're ready for it.
       sendToMainWindow('dispatchAction', {payload: {link: startupURL}, type: DeeplinksGen.link})
       startupURL = null
-    } else if (!isDarwin && process.argv.length > 1 && process.argv[1].startsWith('web+stellar:')) {
+    } else if (!isDarwin) {
       // Windows and Linux instead store a launch URL in argv.
-      sendToMainWindow('dispatchAction', {payload: {link: process.argv[1]}, type: DeeplinksGen.link})
+      let link: string | null = null
```
Comment (cjb) [inline shared/desktop/app/node.desktop.tsx]: I think `let link = ''` is fine, it'll be falsey if empty

**Comment by eli-schwartz** [comment]
> I noticed there's another codepath here. Technically, this should work even if Keybase isn't running already. But I was unable to figure out how to pass in the argument to the electron binary from run_keybase (or in your case keybase-gui) when starting the GUI via systemd. So this flow doesn't work when clicking links on a website, but it will if you manually run `electron path-to-app keybase://xxx`, regardless if the GUI is already up or not.

I tested with the debian prebuilt binary, version 4.3.0-20190812143918+f8de90a10f -- and keybase://chat/username only worked when invoking 
```
/opt/keybase/Keybase &
/opt/keybase/Keybase keybase://chat/username
```

So I assumed it wasn't implemented for initial startup at all. If it is implemented for that, then, well, I can just let arguments mean "invoke directly instead of with systemd".

**Comment by heronhaye** [comment]
There was a bug in the prebuilt binary where it was only accepting web+stellar links but not keybase links - that is fixed in the second commit I added.

**Comment by eli-schwartz** [comment]
Oh, so if you were fixing it for the prebuilt binary too then that makes sense. :)

And indeed, I updated my distro build to use the combined patch on master, and it works for the first start. And nuking /opt/keybase/resources/app and cp'ing /usr/share/keybase-app into /opt/keybase/resources/app allows the (now frankensteined) debian prebuilt binary to accept links on startup too.

Thanks for clarifying.

**Comment by heronhaye** [comment]
Great, sounds good. Just to be clear, that won't work for links you click in your browser like keybase://public/eschwartz, right? Since the registered handler is the desktop file `/usr/share/applications/keybase.desktop`, which I presume runs `keybase-gui %u`?

**Comment by eli-schwartz** [comment]
It does work in Firefox, since the keybase-gui script will, if it sees arguments, simply execute `electron /usr/share/keybase/ "$@"`.

**Pull request merged**